### DISCUSSION TITLE
Error during git checkout to edge branch

### DISCUSSION THREAD

**Discussion by wael-t** [FreshRSS/FreshRSS#5991] [Q&A] [answered]
**Describe the bug**
When I try to update through the GUI, I am getting this message: " The update process has encountered an error: Git error: Error during git checkout to edge branch. Please change branch manually! ❌"

**To Reproduce**
1. Go to my FreshRSS domain URL
2. Click on the settings gear
3. Scroll down to update and click on it
4. click on start update
5. I get this message: " The update process has encountered an error: Git error: Error during git checkout to edge branch. Please change branch manually! ❌" 

**Expected behavior**
A normal update to the app when I click on start update.

**Screenshots**
![Screenshot from 2023-12-25 16-48-25](https://github.com/FreshRSS/FreshRSS/assets/78669714/db9eae3a-c65c-4450-8c35-5e19bbe7fb21)

**Environment information (please complete the following information):**
- Device: Thinkpad X13 Gen4 AMD
- OS: Fedora 39
- Browser: Firefox 121.0
- FreshRSS version: 1.23.1-dev
- Database version: Postgres 14.10
- PHP version: PHP8.2
- Installation type: VPS Ubuntu 22, nginx.

**Comment by wael-t** [comment]
I tried the suggested solution in a similar issue #3665 reported in 20221, but when I execute the proposed git commands in the /FreshRSS I get `fatal: not a git repository (or any of the parent directories): .git`. 
I am quite sure when I installed it I used the git method!

**Comment by Alkarex** [comment, marked as answer]
@wael-t Try to check your FreshRSS folder:

```sh
cd ./FreshRSS/
ls -alh
git status
```

If there is no `.git/` folder and/or if `git status` fails, it is probably that you installed through another method or that it was deleted or corrupted. But this can be fixed :-)

```sh
rm -fr .git/
git init
git remote add origin https://github.com/FreshRSS/FreshRSS.git
git fetch
git checkout -f -b edge origin/edge
git clean -f -d
```

and then you can resume your normal git operations:

```sh
git pull --ff-only
cli/access-permissions.sh
```

and if you want to re-allow updates through the Web interface (a bit less secure):

```sh
# Debian
chown www-data:www-data -R .
# Alpine
chown apache:www-data -R .
```

**Reply by wael-t** [reply]
Thank a lot for your comments, I did saw a .git directory in my /FreshRSS and it only included another directory named .git 

I proceed to delete ths .git and followed your fix commands (including the permissions part) and now it seems well configured in the /FreshRSS and the 'git status' returns 

'On branch edge
Your branch is up to date with 'origin/edge'.

nothing to commit, working tree clean'

However, now when I go to the GUI on the update page I get the following message: 
"An update of FreshRSS is available (Version git), but check permissions on /usr/share/FreshRSS directory. HTTP server must have have write permission." 

I did the update permission  'chmod -R g+w .' inside /FreshRSS and went back to the GUI update page and all the sudden I am now facing again the initial problem receiving the following message reported in my original post "The update process has encountered an error: Git error: Error during git checkout to edge branch. Please change branch manually!". Am not sure why this is happening!

**Reply by Alkarex** [reply]
Ah, this is due to git 2.35.2+ being stricter about ownership checks.
https://github.blog/2022-04-18-highlights-from-git-2-36/#stricter-repository-ownership-checks
Changing group permission is not enough, and you must change ownership.
Try the following (note: this is only for people interested in updating through the Web interface, as it slightly decreases security):

```sh
# Debian
chown www-data:www-data -R .
# Alpine
chown apache:www-data -R .
```

We should update our documentation. PR welcome:
* https://github.com/FreshRSS/FreshRSS/blob/edge/docs/en/admins/06_LinuxInstall.md#part-2-installing-freshrss
* https://github.com/FreshRSS/FreshRSS/blob/edge/README.fr.md#exemple-dinstallation-compl%C3%A8te-sur-linux-debianubuntu

**Reply by wael-t** [reply]
Great! This does resolve the issue completely and the update went flawlessly. Thanks a lot for you feedback.

**Reply by Alkarex** [reply]
PR done https://github.com/FreshRSS/FreshRSS/pull/5995
